### PR TITLE
Rename `all-packages` to `package *`

### PR DESCRIPTION
**Pull request by angerman**
Using `package *` for all packages is in line with `package foo` for
and does not introduce a new keyword to remember.

Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog.
* [x] The documentation has been updated, if necessary.
* [x] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: cabal-install/Distribution/Client/ProjectConfig/Legacy.hs [hunk 1]
```diff
@@ -81,7 +81,7 @@ import Distribution.Simple.Command
          , OptionField, option, reqArg' )
 
 import qualified Data.Map as Map
-
+import Debug.Trace
```
Comment (ezyang) [inline cabal-install/Distribution/Client/ProjectConfig/Legacy.hs]: Kill me

**Comment by ezyang** [comment]
Good thing we fixed this before release, otherwise I'd tell you that you have to support the old syntax too XD

**Comment by ezyang** [comment]
There needs to be a test.

**Comment by angerman** [comment]
> Good thing we fixed this before release, otherwise I'd tell you that you have to support the old syntax too XD
Right, that's why I tried to wedge it in as early as possible.

> There needs to be a test.
There are already tests from  #4972 in the testsuite (https://github.com/haskell/cabal/pull/4972/files#diff-a047033a96ad8af1f66c9061abc0aca5), what kind of additional tests do you envision?

**Comment by ezyang** [comment]
I looked in the diff and didn't see any test files which got changed from `all-packages` to `package *`, so I assumed that there is no test exercising this.

**Comment by 23Skidoo** [comment]
Yep, @dcoutts never added them and we decided to merge that PR anyway lest it bitrot.

**Comment by angerman** [comment]
The current existing tests are only the quick check round trip tests. Which were quite helpful when implementing this change as.

What kind of additional tests do we need?

**Comment by angerman** [comment]
The linux-7.8.4 failure looks like Travis fails to find and install ghc and cabal via apt-get. Maybe it’s dead?

**Comment by 23Skidoo** [comment]
This happens sometimes, rerunning the build usually helps.

**Comment by angerman** [comment]
@23Skidoo I can't restart the Downstream Travis / haskell-pushbot jobs, who can do that?

**Comment by 23Skidoo** [comment]
@angerman It's an open issue, see #4964. Right now I think only @ezyang has access.

**Comment by angerman** [comment]
@23Skidoo so should I just go ahead and merge this one?

**Comment by ezyang** [comment]
I restarted all of the Travis builds by closing/reopening the PR.

**Comment by angerman** [comment]
@ezyang it's the `Downstream Travis` one. And I'm afraid closing and reopening won't help :(
```
+sudo apt-get update
Ign:1 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 InRelease
Hit:2 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 Release
Hit:3 http://apt.postgresql.org/pub/repos/apt trusty-pgdg InRelease
Ign:4 http://dl.google.com/linux/chrome/deb stable InRelease
Hit:6 http://dl.google.com/linux/chrome/deb stable Release
Hit:9 http://security.ubuntu.com/ubuntu trusty-security InRelease
Get:10 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty InRelease [15.4 kB]
Hit:11 https://download.docker.com/linux/ubuntu trusty InRelease
Hit:12 https://dl.hhvm.com/ubuntu trusty InRelease
Get:8 http://dl.bintray.com/apache/cassandra 39x InRelease [3,168 B]
Get:14 http://dl.bintray.com/apache/cassandra 39x/main amd64 Packages [682 B]
Hit:16 https://packagecloud.io/computology/apt-backport/ubuntu trusty InRelease
Get:15 http://dl.bintray.com/apache/cassandra 39x/main i386 Packages [682 B]
Ign:17 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty InRelease
Hit:18 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates InRelease
Hit:19 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports InRelease
Hit:20 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty Release
Hit:21 https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease
Hit:22 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty InRelease
Ign:5 http://toolbelt.heroku.com/ubuntu ./ InRelease
Err:25 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty InRelease
  Could not connect to ppa.launchpad.net:80 (91.189.95.83), connection timed out
Err:26 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty InRelease
  Unable to connect to ppa.launchpad.net:http:
Err:27 http://ppa.launchpad.net/hvr/ghc/ubuntu trusty InRelease
  Unable to connect to ppa.launchpad.net:http:
Err:28 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty InRelease
  Unable to connect to ppa.launchpad.net:http:
Err:29 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty InRelease
  Unable to connect to ppa.launchpad.net:http:
Err:30 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty InRelease
  Unable to connect to ppa.launchpad.net:http:
Ign:31 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main amd64 Packages
Ign:32 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main i386 Packages
Ign:33 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main Translation-en
Ign:31 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main amd64 Packages
Ign:32 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main i386 Packages
Ign:33 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main Translation-en
Err:31 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main amd64 Packages
  Unable to connect to ppa.launchpad.net:http:
Err:32 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main i386 Packages
  Unable to connect to ppa.launchpad.net:http:
Err:33 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty/main Translation-en
  Unable to connect to ppa.launchpad.net:http:
Get:24 http://toolbelt.heroku.com/ubuntu ./ Release [1,609 B]
Get:34 http://toolbelt.heroku.com/ubuntu ./ Release.gpg [473 B]
Get:35 http://toolbelt.heroku.com/ubuntu ./ Packages [636 B]
Fetched 22.7 kB in 10s (2,151 B/s)
Reading package lists... Done
W: Failed to fetch http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/InRelease  Could not connect to ppa.launchpad.net:80 (91.189.95.83), connection timed out
W: Failed to fetch http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/hvr/ghc/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/openjdk-r/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/pollinate/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/webupd8team/java/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/binary-amd64/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/binary-i386/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/i18n/Translation-en  Unable to connect to ppa.launchpad.net:http:
W: Some index files failed to download. They have been ignored, or old ones used instead.
+travis_retry sudo apt-get install --force-yes cabal-install-head cabal-install-2.0 happy-1.19.5 alex-3.1.7 ghc-7.8.4-prof ghc-7.8.4-dyn
+sudo apt-get install --force-yes cabal-install-head cabal-install-2.0 happy-1.19.5 alex-3.1.7 ghc-7.8.4-prof ghc-7.8.4-dyn
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package cabal-install-head
E: Unable to locate package cabal-install-2.0
E: Couldn't find any package by glob 'cabal-install-2.0'
E: Couldn't find any package by regex 'cabal-install-2.0'
E: Unable to locate package happy-1.19.5
E: Couldn't find any package by glob 'happy-1.19.5'
E: Couldn't find any package by regex 'happy-1.19.5'
E: Unable to locate package alex-3.1.7
E: Couldn't find any package by glob 'alex-3.1.7'
E: Couldn't find any package by regex 'alex-3.1.7'
E: Unable to locate package ghc-7.8.4-prof
E: Couldn't find any package by glob 'ghc-7.8.4-prof'
E: Couldn't find any package by regex 'ghc-7.8.4-prof'
E: Unable to locate package ghc-7.8.4-dyn
E: Couldn't find any package by glob 'ghc-7.8.4-dyn'
E: Couldn't find any package by regex 'ghc-7.8.4-dyn'
+sleep 1
+sudo apt-get install --force-yes cabal-install-head cabal-install-2.0 happy-1.19.5 alex-3.1.7 ghc-7.8.4-prof ghc-7.8.4-dyn
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package cabal-install-head
E: Unable to locate package cabal-install-2.0
E: Couldn't find any package by glob 'cabal-install-2.0'
E: Couldn't find any package by regex 'cabal-install-2.0'
E: Unable to locate package happy-1.19.5
E: Couldn't find any package by glob 'happy-1.19.5'
E: Couldn't find any package by regex 'happy-1.19.5'
E: Unable to locate package alex-3.1.7
E: Couldn't find any package by glob 'alex-3.1.7'
E: Couldn't find any package by regex 'alex-3.1.7'
E: Unable to locate package ghc-7.8.4-prof
E: Couldn't find any package by glob 'ghc-7.8.4-prof'
E: Couldn't find any package by regex 'ghc-7.8.4-prof'
E: Unable to locate package ghc-7.8.4-dyn
E: Couldn't find any package by glob 'ghc-7.8.4-dyn'
E: Couldn't find any package by regex 'ghc-7.8.4-dyn'
+sleep 2
+sudo apt-get install --force-yes cabal-install-head cabal-install-2.0 happy-1.19.5 alex-3.1.7 ghc-7.8.4-prof ghc-7.8.4-dyn
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package cabal-install-head
E: Unable to locate package cabal-install-2.0
E: Couldn't find any package by glob 'cabal-install-2.0'
E: Couldn't find any package by regex 'cabal-install-2.0'
E: Unable to locate package happy-1.19.5
E: Couldn't find any package by glob 'happy-1.19.5'
E: Couldn't find any package by regex 'happy-1.19.5'
E: Unable to locate package alex-3.1.7
E: Couldn't find any package by glob 'alex-3.1.7'
E: Couldn't find any package by regex 'alex-3.1.7'
E: Unable to locate package ghc-7.8.4-prof
E: Couldn't find any package by glob 'ghc-7.8.4-prof'
E: Couldn't find any package by regex 'ghc-7.8.4-prof'
E: Unable to locate package ghc-7.8.4-dyn
E: Couldn't find any package by glob 'ghc-7.8.4-dyn'
E: Couldn't find any package by regex 'ghc-7.8.4-dyn'
The command "./travis-install.sh" failed and exited with 100 during .
```
Travis has the following yellow warning attached as well:
```
 This job ran on our Trusty environment, which has been updated on Tuesday, December 12th, 2017. Read all about this on our blog and take note that you can add group: deprecated-2017Q4 in your .travis.yml file to use the previous image version.
```

**Comment by angerman** [comment]
@ezyang see https://travis-ci.org/haskell-pushbot/cabal-binaries/builds/330687892

**Comment by ezyang** [comment]
The downstream Travis builds are triggered by the regular Travis build, so if the regular one restarts, the downstream ones restart too.

The real problem seems to be that our Travis queue is completely hosed right now. Since Travis claims there's no outage right now, I'm just going to cancel all ongoing jobs and see if that unsticks it.

**Comment by angerman** [comment]
```
$ env CABAL_INSTALL_MONOLITHIC_MODE=SolverQuickCheck ./cabal --quickcheck-tests=1000
Solver QuickCheck
  UnitTests.Distribution.Solver.Modular.QuickCheck
    target and goal order do not affect solvability:                           cabal: internal error: Unable to commit 1048576 bytes of memory
    (GHC version 8.2.2 for x86_64_unknown_linux)
    Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```
:(

**Comment by 23Skidoo** [comment]
Closed & reopened to restart CI. OOM issue is resolved in `master`, I think.

**Comment by 23Skidoo** [comment]
This is finally green, merging.

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Pull request merged**